src/test/regress/sql/executor_local_failure.sql

```diff
@@ -0,0 +1,30 @@
+CREATE SCHEMA failure_local_modification;
+SET search_path TO failure_local_modification;
+SET citus.next_shard_id TO 1989000;
+
+SET citus.shard_replication_factor TO 1;
+CREATE TABLE failover_to_local (key int PRIMARY KEY, value varchar(10));
+SELECT create_reference_table('failover_to_local');
+
+\c - - - :worker_2_port
+
+SET search_path TO failure_local_modification;
+
+-- prevent local connection establishment, imitate
+-- a failure
+ALTER SYSTEM SET citus.local_shared_pool_size TO -1;
+SELECT pg_reload_conf();
+SELECT pg_sleep(0.2);
+BEGIN;
+	-- we force the execution to use connections (e.g., remote execution)
+	-- however, we do not allow connections as local_shared_pool_size=-1
+	SET LOCAL citus.enable_local_execution TO false;
+	INSERT INTO failover_to_local VALUES (1,'1'), (2,'2'),(3,'3'),(4,'4');
+ROLLBACK;
+
+-- although we ingested 4 rows, we cannot see them
+SELECT count(*) FROM failover_to_local;
```

Comment:
This is guaranteed to give zero rows as we are calling ROLLBACK. Customer scenario is, we are doing a COMMIT and "quietly" failing to INSERT. I think we should change ROLLBACK to COMMIT?

Reply:
Yeah, that's right, it is left over from the initial commit, where we can show the bug in action. I'll change the test